lib/client/user/user-client.js

```diff
@@ -182,78 +200,88 @@ UserClient.prototype.auth = function(username, password) {
-      deferred.resolve(true);
-    } else {
-      deferred.resolve(false);
-    }
+
+  return this.initPromise.then(function() {
+    var deferred = q.defer();
+    $fh.auth.hasSession(function(err, exists) {
+      if (err) {
+        deferred.reject(new Error(err));
+      } else if (exists) {
+        //user is already authenticated
+        //optionally we can also verify the session is acutally valid from client. This requires network connection.
+        deferred.resolve(true);
+      } else {
+        deferred.resolve(false);
+      }
+    });
+    return deferred.promise;
   });
-  return deferred.promise;
 };
 
 UserClient.prototype.clearSession = function() {
-  var deferred = q.defer();
   var self = this;
 
-  self.userVerified = false;
-  $fh.auth.clearSession(function(err) {
-    if (err) {
-      console.log('Failed to clear session: ', err);
-      deferred.reject(new Error(err));
-    } else {
-      self.mediator.publish('wfm:auth:profile:change', null);
+  return this.initPromise.then(function() {
+    self.userVerified = false;
+    return self.xhr({
+      path: config.authpolicyPath + "/revokesession",
+      method: "POST",
+      body: {}
+    }).then(function() {
       localStorage.clear();
-      deferred.resolve(true);
-    }
-
+      self.mediator.publish('wfm:auth:profile:change', null);
+    });
   });
-  return deferred.promise;
 };
 
-UserClient.prototype.verify = function() {
+UserClient.prototype.verify = function(forceVerify) {
   var self = this;
 
   //If the user has already been verified, then no need to do it again.
-  if (this.userVerified) {
+  if (this.userVerified && !forceVerify) {
     return q.when(true);
   }
 
-  var deferred = q.defer();
+  return this.initPromise.then(function() {
+    return self.xhr({
+      path: config.authpolicyPath + "/verifysession",
+      method: "POST",
+      body: {}
+    }).then(function(validationResult) {
+      self.userVerified = validationResult && validationResult.isValid;
+      return self.userVerified;
+    });
 
-  $fh.auth.verify(function(err, valid) {
-    if (err) {
-      console.log('failed to verify session');
-      deferred.reject(new Error(err));
-      return;
-    } else if (valid) {
-      self.userVerified = true;
-      deferred.resolve(true);
-    } else {
-      deferred.resolve(false);
-    }
   });
-  return deferred.promise;
 };
 
 UserClient.prototype.getProfile = function() {
   return q.when(retrieveProfileData());
 };
 
-module.exports = function(mediator) {
+module.exports = function(mediator, config) {
+  config = config || {};
 
   //Only want a single user client per application,
   //If one already exists, use this one.
   if (userClient) {
     return userClient;
   }
 
+  //If the session should be verified every time the app is resumed
+  //Then the http request should be sent.
+  if (config.forceSessionVerificationOnResume) {
+    document.addEventListener("resume", function() {
```

Comment:
As per documentation if we register this before device is ready this will have no effect.
Better to ensure that deviceready event was triggered in ready function.
See: https://cordova.apache.org/docs/en/5.1.1/cordova/events/events.resume.html